Updated FAQ and WAL dump/encryption based on AA feedback

Andriciuc opened the pull request:
- updated FAQ with WAL encryption notes about it's beta status
- Added links to How does pg_tde make my data safe? FAQ for KMS chapters we added
- updated pg_waldump with a note about wal encryption beta status
- Removed post quantum question from FAQ

Andriciuc committed:
Updated FAQ and WAL dump/encryption based on AA feedback

- updated FAQ with WAL encryption notes about it's beta status
- Added links to How does pg_tde make my data safe? FAQ for KMS chapters we added
- updated pg_waldump with a note about wal encryption beta status
- Added

dutow commented on contrib/pg_tde/documentation/docs/faq.md:
```diff
@@ -70,13 +74,14 @@ Thus, to protect your sensitive data, consider using TDE to encrypt it at the ta
 
-HashiCorp Vault can also act as the KMIP server, managing cryptographic keys for clients that use the KMIP protocol.
+!!! note
+    HashiCorp Vault can also act as a KMIP server, managing cryptographic keys for clients that use the KMIP protocol.
```
only the enterprise (paid) version

Andriciuc replied:
Added: *(KMIP functionality is available in Vault's enterprise edition.)* after the end paragraph

dutow commented on contrib/pg_tde/documentation/docs/faq.md:
```diff
@@ -114,10 +122,6 @@ For WAL encryption, AES-CTR-128 is used.
 
 The support of other encryption mechanisms such as AES256 is planned for future releases. Reach out to us with your requirements and usage scenarios of other encryption methods are needed.
 
-## Is post-quantum encryption supported?
```
We still need to answer this question

Andriciuc replied:
Hmm, okay, readded this, but left it simple: No. Post-quantum encryption is not currently supported.

Andriciuc committed:
Update faq.md

readded post-quantum encryption question and added a small disclaimer for KMIP HashiCorp that it is available for the enterprise edition.

Andriciuc committed:
Merge branch 'release-17.5.2' into doc-improvements-based-on-AA-feedback

Andriciuc committed:
Merge branch 'release-17.5.2' into doc-improvements-based-on-AA-feedback

Andriciuc committed:
Merge branch 'release-17.5.2' into doc-improvements-based-on-AA-feedback

Andriciuc committed:
Merge branch 'release-17.5.2' into doc-improvements-based-on-AA-feedback

AndersAstrand reviewed (approved):
Please squash before committing

Andriciuc committed:
Merge branch 'release-17.5.2' into doc-improvements-based-on-AA-feedback
